jena-arq/src/main/java/org/apache/jena/sparql/exec/tracker/ChainingQueryDispatcherExecTracker.java

```diff
@@ -0,0 +1,32 @@
+package org.apache.jena.sparql.exec.tracker;
+
+import org.apache.jena.query.Query;
+import org.apache.jena.query.Syntax;
+import org.apache.jena.sparql.core.DatasetGraph;
+import org.apache.jena.sparql.engine.binding.Binding;
+import org.apache.jena.sparql.engine.dispach.ChainingQueryDispatcher;
+import org.apache.jena.sparql.engine.dispach.QueryDispatcher;
+import org.apache.jena.sparql.exec.QueryExec;
+import org.apache.jena.sparql.util.Context;
+
+public class ChainingQueryDispatcherExecTracker
+    implements ChainingQueryDispatcher
+{
+    @Override
+    public QueryExec create(Query query, DatasetGraph dsg, Binding initialBinding, Context context,
+            QueryDispatcher chain) {
+        QueryExec delegate = chain.create(query, dsg, initialBinding, context);
+        QueryExec result = TaskTrackerRegistry.track(context, delegate);
+        TaskTrackerRegistry.remove(context);
+        return result;
+    }
```

Comment:
Relevant code for intercepting query execution construction over a dataset graph using the newly proposed dispatcher system. Here, execution tracking is added.